zerver/lib/topic_mutes.py

```diff
@@ -134,3 +134,20 @@ def is_muted(recipient_id: int, topic: str) -> bool:
         return (recipient_id, topic.lower()) in tups
 
     return is_muted
+
+
+def get_users_muted_topic_data(topic_name: str, stream_id: int) -> List[Dict[str, Any]]:
+    rows = MutedTopic.objects.filter(stream_id=stream_id, topic_name__iexact=topic_name,).values(
+        "user_profile",
+        "recipient_id",
+        "date_muted",
+    )
+
+    return [
+        dict(
+            user_profile=UserProfile.objects.get(id=row["user_profile"]),
```

Comment:
We should never fetch UserProfile objects in a loop like this, as it results in poor database performance.  I'm also not convinced you need it; shouldn't the ID be good enough for where we use these fields?

Reply:
Okay, maybe I can use `get_user_profile_by_id` function in `zerver.models.py` ?
`ID` is not good enough as functions: `remove_topic_mute` and `add_topic_mute` require `UserProfile` instead of `ID`.

Reply:
No, `get_user_profile_by_id` does a cache query. We should do a bulk fetch, e.g. by adding `select_related` to the query here.